sdks/java/javadoc/build.gradle

```diff
@@ -26,64 +26,33 @@
 description = "Apache Beam :: SDKs :: Java :: Aggregated Javadoc"
 apply plugin: 'java'
 
-def exportedJavadocProjects = [
-  ':beam-runners-apex',
-  ':beam-runners-core-construction-java',
-  ':beam-runners-core-java',
-  ':beam-runners-java-fn-execution',
-  ':beam-runners-local-java-core',
-  ':beam-runners-direct-java',
-  ':beam-runners-reference-java',
-  ':beam-runners-flink_2.11',
-  ':beam-runners-google-cloud-dataflow-java',
-  ':beam-runners-spark',
-  ':beam-runners-gearpump',
-  ':beam-sdks-java-core',
-  ':beam-sdks-java-fn-execution',
-  ':beam-sdks-java-extensions-google-cloud-platform-core',
-  ':beam-sdks-java-extensions-join-library',
-  ':beam-sdks-java-extensions-json-jackson',
-  ':beam-sdks-java-extensions-protobuf',
-  ':beam-sdks-java-extensions-sketching',
-  ':beam-sdks-java-extensions-sorter',
-  ':beam-sdks-java-extensions-sql',
-  ':beam-sdks-java-harness',
-  ':beam-sdks-java-io-amazon-web-services',
-  ':beam-sdks-java-io-amqp',
-  ':beam-sdks-java-io-cassandra',
-  ':beam-sdks-java-io-elasticsearch',
-  ':beam-sdks-java-io-elasticsearch-tests-2',
-  ':beam-sdks-java-io-elasticsearch-tests-5',
-  ':beam-sdks-java-io-elasticsearch-tests-6',
-  ':beam-sdks-java-io-elasticsearch-tests-common',
-  ':beam-sdks-java-io-google-cloud-platform',
-  ':beam-sdks-java-io-hadoop-common',
-  ':beam-sdks-java-io-hadoop-file-system',
-  ':beam-sdks-java-io-hadoop-input-format',
-  ':beam-sdks-java-io-hbase',
-  ':beam-sdks-java-io-hcatalog',
-  ':beam-sdks-java-io-jdbc',
-  ':beam-sdks-java-io-jms',
-  ':beam-sdks-java-io-kafka',
-  ':beam-sdks-java-io-kinesis',
-  ':beam-sdks-java-io-mongodb',
-  ':beam-sdks-java-io-mqtt',
-  ':beam-sdks-java-io-parquet',
-  ':beam-sdks-java-io-redis',
-  ':beam-sdks-java-io-solr',
-  ':beam-sdks-java-io-tika',
-  ':beam-sdks-java-io-xml',
-]
+for (String projName : rootProject.ext.allProjectNames) {
+  if (!projName.equals(':' + project.name) && !projName.equals(':beam-sdks-java-bom')) {
+    evaluationDependsOn(projName)
```

Comment:
The opposite of your first sentence is what is happening - this project depends on every other project. But anyway, the rest of what you say seems true, you can encounter cycles. I actually did encounter a cycle because `beam-sdks-java-bom` also uses `evaluationDependsOn` for all modules. When there is a cycle, you just need to determine the order of precedence. My decision was all modules -> javadoc -> bom. That is why if you look down further in this file, you will see that `:beam-sdks-java-bom` is skipped. Thus, you can never have javadoc for the bom, but the bom can include the javadoc module.

Reply:
Got it, thanks for the clarification. I'm ok leaving this for now.